searx/query.py

```diff
@@ -63,100 +70,225 @@ def _parse_query(self):
 
             # this force the timeout
             if query_part[0] == '<':
```

Comment:
Maybe create a helper function `_is_timeout_setting` so the ugly check can be hidden:

```python
if _is_timeout_setting(query_part):
    # set timeout whatever
```

Helper func:
```python
def _is_timeout_setting(query_part):
    return query_part[0] == '<'
```

Reply:
There is a now a `QueryPartParser` class, and subclasses : `TimeoutParser`, `LanguageParser`, `ExternalBangParser`, `BangParser`.

Each `QueryPartParser`:
* has a class method `check(raw_value)` to check if the raw_value can be parsed.
* has a method `__call__(self, raw_value)` to actual parse the raw_value. The method may return False (for example `!this is not a bang`)

`RawTextQuery._parse_query` becomes very simple.

`RawTextQuery.PARSER_CLASSES` allows a plugin to add some other parsers (the information won't be sent to the engine, but that's a start).